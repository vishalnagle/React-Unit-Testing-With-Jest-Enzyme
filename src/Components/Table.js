import React from "react";

export default class Table extends React.Component{

    constructor(props){
        super(props)
    }


    render(){
        return (<table className="custom-border">
            <thead>
                <tr>
                    <th>Name</th>
                    <th>Email</th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <td>Demo</td>
                    <td>NA</td>
                </tr>
            </tbody>
        </table>)
    }

}
