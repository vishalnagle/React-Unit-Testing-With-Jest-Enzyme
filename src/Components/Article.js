import React from "react";
import Table from "./Table";

export default class Article extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      count: 0,
    };
  }

  render() {
    const { count } = this.state;
    return (
      <div>
        <div className={`clicks ${count}`}>{count} clicks</div>
        <button onClick={()=>{this.setState({count:count+1})}}>
            Increment
        </button>
        <h1>Article Details</h1>
        <Table/>
      </div>
    );
  }
}
