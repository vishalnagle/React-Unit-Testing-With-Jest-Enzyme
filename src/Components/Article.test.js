import { configure } from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";
configure({ adapter: new Adapter() });
// import About from "./About.js";
import Article from "./Article";
import { shallow,mount } from "enzyme";
import Table from "./Table";

// Simulate test case tutorial

describe("Article component", () => {
  it("click count", () => {
    let wrapper = shallow(<Article />);
    expect(wrapper.find(".clicks-0").length).toEqual(1);
    wrapper.find("button").simulate("click");
    expect(wrapper.find(".clicks-0").length).toEqual(1);
  });

  it("click 3 count", () => {
    let wrapper = shallow(<Article />);
    expect(wrapper.find(".clicks-0").length).toEqual(1);
    wrapper.find("button").simulate("click");
    expect(wrapper.find(".clicks-2").length).toEqual(1);
  });

  it("dive", () => {
    let wrapper = shallow(<Article />);
    expect(wrapper.find(Table).dive().find(".custom-border").length).toBe(1);
  });

  // Match snapshot

  it("snapshot 1", () => {
    let wrapper = shallow(<Table />);
    expect(wrapper).toMatchSnapshot();
  });

  test("snapshot 2", () => {
    let wrapper = shallow(<Table />);
    expect(wrapper).toMatchSnapshot();
  });
});

// Mount Full Rendering
// mount vs shallow

describe("mount vs shallow",()=>{
    test("snapshot 3", () => {
       let wrapper = shallow(<Article/>);
       console.log("shallow wrapper",wrapper.debug())

    //    mount
    let wrapper1 = mount(<Article/>);
       console.log("mount wrapper",wrapper1.debug())
      });
})
