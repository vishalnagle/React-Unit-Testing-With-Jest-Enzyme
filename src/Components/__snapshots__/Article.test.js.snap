// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component snapshot 1 1`] = `ShallowWrapper {}`;

exports[`Article component snapshot 2 1`] = `ShallowWrapper {}`;
